executor/admin.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/ranger"
+	"github.com/pingcap/tipb/go-tipb"
+	"golang.org/x/net/context"
+)
+
+var _ Executor = new(CheckIndexRangeExec)
+
+// CheckIndexRangeExec output the index values which has handle between begin and end.
+type CheckIndexRangeExec struct {
+	baseExecutor
+
+	table    table.Table
+	index    table.Index
+	is       infoschema.InfoSchema
+	startKey []types.Datum
+
+	handleRanges []ast.HandleRange
+	srcChunk     *chunk.Chunk
+
+	result   distsql.SelectResult
+	partial  distsql.PartialResult
+	typesMap map[int64]*types.FieldType
+	cols     []*model.ColumnInfo
+}
+
+// NextChunk implements the Executor Next interface.
+func (e *CheckIndexRangeExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	handleIdx := e.schema.Len() - 1
+	for {
+		err := e.result.NextChunk(ctx, e.srcChunk)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if e.srcChunk.NumRows() == 0 {
+			return nil
+		}
+		for i := 0; i < e.srcChunk.NumRows(); i++ {
+			row := e.srcChunk.GetRow(i)
+			handle := row.GetInt64(handleIdx)
+			for _, hr := range e.handleRanges {
+				if handle >= hr.Begin && handle < hr.End {
+					chk.AppendRow(row)
+				}
+			}
+		}
+		if chk.NumRows() >= e.maxChunkSize {
```

Comment:
move this after we appending a row into `chk` in line 68

Reply:
This is simpler, for admin statement, we don't need to strictly limit the chunk size.

Reply:
I think since we have a rule like "chunk.NumRows() should not exceed max chunk size", we should respect it. Beside it won't bring too much complexity.